astroquery/svo_fps/core.py

```diff
@@ -0,0 +1,115 @@
+import numpy as np
+import requests
+import io
+
+from astropy import units as u
+from astropy.io.votable import parse_single_table
+# The VOTables fetched from SVO contain only single table element, thus parse_single_table
+
+from ..query import BaseQuery
+
+__all__ = ['SvoFpsClass', 'SvoFps']
+
+FLOAT_MAX = np.finfo(np.float64).max
+
+
+class SvoFpsClass(BaseQuery):
+    SVO_MAIN_URL = 'http://svo2.cab.inta-csic.es/theory/fps/fps.php'
+
+    def data_from_svo(self, query, cache=True,
+                      error_msg='No data found for requested query'):
+        """Get data in response to the query send to SVO FPS
+
+        Parameters
+        ----------
+        query : dict
+            Used to create a HTTP query string i.e. send to SVO FPS to get data.
+            In dictionary, specify keys as search parameters (str) and
+            values as required. List of search parameters can be found at
+            http://svo2.cab.inta-csic.es/theory/fps/fps.php?FORMAT=metadata
+        error_msg : str, optional
+            Error message to be shown in case no table element found in the
+            responded VOTable. Use this to make error message verbose in context
+            of the query made (default is 'No data found for requested query')
+        cache : bool
+            Cache the results?  Defaults to True
+
+        Returns
+        -------
+        astropy.table.table.Table object
+            Table containing data fetched from SVO (in response to query)
+        """
+        response = self._request("GET", self.SVO_MAIN_URL, params=query,
+                                 cache=cache)
+        response.raise_for_status()
+        votable = io.BytesIO(response.content)
+        try:
+            return parse_single_table(votable).to_table()
+        except IndexError:
+            # If no table element found in VOTable
+            raise IndexError(error_msg)
+
+    def get_filter_index(self, wavelength_eff_min=0*u.angstrom,
+                         wavelength_eff_max=FLOAT_MAX*u.angstrom, **kwargs):
+        """Get master list (index) of all filters at SVO
+        Optional parameters can be given to get filters data for specified
+        Wavelength Effective range from SVO
+
+        Parameters
+        ----------
+        wavelength_eff_min : `~astropy.units.Quantity` having units of angstrom, optional
+            Minimum value of Wavelength Effective (default is 0 angstrom)
+        wavelength_eff_max : `~astropy.units.Quantity` having units of angstrom, optional
+            Maximum value of Wavelength Effective (default is a very large
+            quantity FLOAT_MAX angstroms i.e. maximum value of np.float64)
+
```

Comment:
I generally don't like dragging around a bunch of copy-pasted docs since that's a way to propagate errors / create maintenance burden. I thought we had a generic docstring tagger that could add this stuff, but I forget what it is and where it lives...

Reply:
It's part of the async_to_sync decorator... can it be called separately? That would definitely be handy.

Reply:
added by hand =/